src/vos/vos_io.c

```diff
@@ -419,6 +417,20 @@ vos_ioc_reserve_init(struct vos_io_context *ioc)
 		return -DER_NOMEM;
 
 	ioc->ic_rsrvd_scm->rs_actv_cnt = total_acts;
+
+	if (!dtx_is_valid_handle(dth) || dth->dth_deferred == NULL)
+		return 0;
+
+	/** Reserve enough space for any deferred actions */
```

Comment:
Some confused, my understand is that the "dth_deferred[i]" will be used as output parameter for pmemobj_defer_free(), one defer_free() only need one "pobj_action". Since we have already created N (N is the 'dth_modification_cnt') "vos_rsrvd_scm" slots when DTX handle init, then for each slot, we only need to create one "pobj_action", right?  Do we really need so large size to hold multiple "pobj_action"? Or I missed anything?